### PR TITLE
Watch checkup job

### PR DESCRIPTION
**Pull request by ormergi**
Currently the checkup-framework launches a checkup and
exit,in order to do proper teardown, dumping logs and provide
meaningful feedback regarding the checkup status,
it is necessary to watch the Job status during execution,
wait for it to finish.

Signed-off-by: Or Mergi <ormergi@redhat.com>

### CODE REVIEW

File: hack/run-echo-checkup [hunk 1]
```diff
@@ -46,3 +48,8 @@ kubectl delete -f $USER_CONFIG_MANIFEST_FILE
 echo
 date +"%Y-%m-%dT%H:%M:%S%:z" | tr -d "\n"
 echo " *** Done ***"
+
+if [ "$job_failed" == "true" ]; then
+  echo "FATAL: error occurred while running echo checkup" 2>&1
```
Comment (orelmisan) [inline hack/run-echo-checkup:49]: This should go to stderr
1>&2

File: hack/deploy-framework [hunk 2]
```diff
@@ -7,10 +7,10 @@ FRAMEWORK_ENV_MANIFEST_FILE="./manifests/k8s-checkup-framework.yaml"
 kubectl version
```
Comment (orelmisan) [inline hack/deploy-framework:7]: This change does not belong in this PR.

File: checkups/echo/build-image [hunk 3]
```diff
@@ -3,6 +3,13 @@
 set -x
```
Comment (orelmisan) [inline checkups/echo/build-image:3]: This change does not belong to this PR.

File: hack/run-echo-checkup [hunk 4]
```diff
@@ -29,8 +27,6 @@ while true; do
     job_failed=true
     break
   fi
-
-  sleep $WAIT_PERIOD_SEC
```
Comment (orelmisan) [inline hack/run-echo-checkup:31]: If you remove this sleep, the loop will run for no reason as fast as it can.

File: pkg/checkup/watch.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8swatch "k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+const JobNameLabel = "job-name"
+
+func WaitForJobToComplete(client *kubernetes.Clientset, jobName, jobNamespace string, timeout time.Duration) (*batchv1.Job, error) {
+	jobLabel := fmt.Sprintf("%s=%s", JobNameLabel, jobName)
+	jobWatcher, err := client.BatchV1().Jobs(jobNamespace).Watch(context.Background(), metav1.ListOptions{LabelSelector: jobLabel})
```
Comment (orelmisan) [inline pkg/checkup/watch.go:20]: AFAIK the usage of Watch() is not recommend.
IMO we should use an informer.

**Review by orelmisan — COMMENTED** [review]
Thanks for your PR.
Please see inline comments.

**Comment by orelmisan** [comment]
Please add a PR description.

File: hack/run-echo-checkup (hunk 4 above)
Comment (ormergi) [reply hack/run-echo-checkup:31]: Oh I got it wrong, I will restore it sorry.

File: hack/run-echo-checkup (hunk 1 above)
Comment (ormergi) [reply hack/run-echo-checkup]: Done

File: pkg/checkup/watch.go (hunk 5 above)
Comment (ormergi) [reply pkg/checkup/watch.go:20]: Would you please share why Watch isnt recommended?

Note that executing a checkup is one shot operation, and there is only one instance of it.
Using an informer for the checkup Job is an overkill, requires some boilerplate and thus have higher maintenance.
For now I prefer to keep it simple for now and use Watch and later on consider switching to Informer.

File: checkups/echo/build-image (hunk 3 above)
Comment (ormergi) [reply checkups/echo/build-image]: I added this commit to this PR because its the closest thing to an end-to-end test.
Using it enables you to verify that running a checkup though the framework actually works.

I will move it to another PR.

File: hack/deploy-framework (hunk 2 above)
Comment (ormergi) [reply hack/deploy-framework]: I will move it to another PR

**Comment by ormergi** [comment]
@orelmisan could you please take another look?

File: hack/deploy-framework (hunk 2 above)
Comment (ormergi) [reply hack/deploy-framework]: Moved to #14

**Comment by orelmisan** [comment]
Thank you!
